src/domain/repositories/UserAuthoritiesRepository.ts

```diff
@@ -0,0 +1,20 @@
+import { User } from "data/d2-users/D2Users.types";
+import {
+    TemplateGroupWithAuthorities,
+    UserMonitoringCountResponse,
+    UserMonitoringDetails,
+    UsersOptions,
+} from "domain/entities/UserMonitoring";
+
+export interface UserAuthoritiesRepository {
+    processUserGroups(
+        options: UsersOptions,
+        completeTemplateGroups: TemplateGroupWithAuthorities[],
+        users: User[]
+    ): Promise<UserMonitoringCountResponse>;
+    processUserRoles(
```

Comment:
[Clean Arc.] Same as before, for the processUserRoles we can extract all the logic into the use case, build the users we want to udpate and then in the usecase call the `UserRepository.save(userToPost);`